### PR TITLE
feat: use ursa-optional for lightning fast key generation

### PR DESCRIPTION
**Pull request by mkg20001**
The difference between ursa and ursa-optional is that ursa-optional does not cause any problems if it fails to compile

### CODE REVIEW

**Comment by diasdavid** [comment]
@dignifiedquire can you verify, merge and release?

**Comment by diasdavid** [comment]
@mkg20001 can you bring coverage up?

**Comment by mkg20001** [comment]
@diasdavid Added tests that run .generateKeyPair() with both crypto libs

**Comment by mkg20001** [comment]
As I neither have a windows machine nor VM currently, somebody else would have to debug whats causing this issue. Most likely the module just failed to compíle.

**Comment by mkg20001** [comment]
@dignifiedquire Could you merge this?

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "protons": "^1.0.1",
     "rsa-pem-to-jwk": "^1.1.3",
     "tweetnacl": "^1.0.0",
+    "ursa-optional": "^0.9.8",
```
Comment (diasdavid) [inline package.json]: This should be failing the linting. Use ~ instead of ^ for < 1.0.0 modules.

**Comment by dignifiedquire** [comment]
Sorry, this got lost in my inbox :(

**Pull request merged**